crescent/utils/get_sig.py

```diff
@@ -33,4 +33,7 @@ def get_parameters(func: Callable[..., Any]) -> Sequence[Parameter]:
 
     sig = signature(func)
 
-    return [convert_signiture(param, type_hints) for param in sig.parameters.values()]
+    return [
+        convert_signiture(param, type_hints)  # pyright: ignore
+        for param in sig.parameters.values()
+    ]
```

Comment:
This seems sketchy.

Reply:
I think pyright complains because it's unknown due to the 3.9 check, type_hints is unknown, but it can be resolved in the else statement where it converts signature for 3.8. So after the if/else block in the return statement, the variable is partially unknown. Maybe we can manually type hint it?

Reply:
At least a comment would be nice. Also, prefer `cast` over type: ignore wherever possible

Reply:
```python
if version_info >= (3, 9):
        type_hints: dict[str, Any] = get_type_hints(func, include_extras=True)  # type: ignore
```
I type hinted type_hints above where it checks for 3.9, so pyright can carry that over to 3.8, where it is being set as the type checking version in settings.